src/catalog/catalog_entry/schema_catalog_entry.cpp

```diff
@@ -102,7 +128,28 @@ CatalogEntry *SchemaCatalogEntry::CreateType(ClientContext &context, CreateTypeI
 CatalogEntry *SchemaCatalogEntry::CreateTable(ClientContext &context, BoundCreateTableInfo *info) {
 	auto table = make_unique<TableCatalogEntry>(catalog, this, info);
 	table->storage->info->cardinality = table->storage->GetTotalRows();
-	return AddEntry(context, move(table), info->Base().on_conflict, info->dependencies);
+	CatalogEntry *entry = AddEntry(context, move(table), info->Base().on_conflict, info->dependencies);
+	if (!entry) {
+		return nullptr;
+	}
+	// add a foreign key constraint in main key table if there is a foreign key constraint
+	vector<unique_ptr<AlterForeignKeyInfo>> fk_arrays;
+	FindForeignKeyInformation(entry, AlterForeignKeyType::AFT_ADD, fk_arrays);
+	for (idx_t i = 0; i < fk_arrays.size(); i++) {
+		// alter primary key table
+		AlterForeignKeyInfo *fk_info = fk_arrays[i].get();
+		catalog->Alter(context, fk_info);
+
+		// make a dependency between this table and referenced table
+		auto &set = GetCatalogSet(CatalogType::TABLE_ENTRY);
+		auto pk_table_entry_ptr = set.GetEntry(context, fk_info->name);
+		if (pk_table_entry_ptr) {
+			info->dependencies.insert(pk_table_entry_ptr);
+		} else {
+			throw ParserException("Can't find table \"%s\" in foreign key constraint", fk_info->name);
```

Comment:
This code does not seem to be triggered - it also should not be a ParserException. ParserExceptions are special in that they allow a transaction to continue. However, it is possible that you have already made changes to the system here (through the `catalog->Alter`) which can then be (incorrectly) committed. Could you add a test case triggering this and turn this into a `CatalogException`?

Reply:
You are right, this code isn't triggered, I will fix it.